File: CS/WindowsFormsApplication21/CustomDataNavigatorButtonCollection.cs
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.NavigatorButtons;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace WindowsFormsApplication21 {
    public class CustomDataNavigatorButtonCollection : DataNavigatorButtonCollection {
        public CustomDataNavigatorButtonCollection(DataNavigatorButtons buttons)
            : base(buttons) {
        }
        protected override void CreateButtons(NavigatorButtonsBase buttons) {
            AddButton(new DataNavigatorFirstButtonHelper(buttons));
            AddButton(new DataNavigatorPrevPageButtonHelper(buttons));
            AddButton(new DataNavigatorPrevButtonHelper(buttons));
            AddButton(new DataNavigatorNextButtonHelper(buttons));
            AddButton(new DataNavigatorNextPageButtonHelper(buttons));
            AddButton(new DataNavigatorLastButtonHelper(buttons));
            AddButton(new DataNavigatorAddButtonHelper(buttons));
            AddButton(new DataNavigatorRemoveButtonHelper(buttons));
            AddButton(new DataNavigatorCancelEditButtonHelper(buttons));
            AddButton(new DataNavigatorEndEditButtonHelper(buttons));
        }
    }
}

File: VB/WindowsFormsApplication21/CustomDataNavigatorButtonCollection.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.NavigatorButtons
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace WindowsFormsApplication21
	Public Class CustomDataNavigatorButtonCollection
		Inherits DataNavigatorButtonCollection
		Public Sub New(ByVal buttons As DataNavigatorButtons)
			MyBase.New(buttons)
		End Sub
		Protected Overrides Sub CreateButtons(ByVal buttons As NavigatorButtonsBase)
			AddButton(New DataNavigatorFirstButtonHelper(buttons))
			AddButton(New DataNavigatorPrevPageButtonHelper(buttons))
			AddButton(New DataNavigatorPrevButtonHelper(buttons))
			AddButton(New DataNavigatorNextButtonHelper(buttons))
			AddButton(New DataNavigatorNextPageButtonHelper(buttons))
			AddButton(New DataNavigatorLastButtonHelper(buttons))
			AddButton(New DataNavigatorAddButtonHelper(buttons))
			AddButton(New DataNavigatorRemoveButtonHelper(buttons))
			AddButton(New DataNavigatorCancelEditButtonHelper(buttons))
			AddButton(New DataNavigatorEndEditButtonHelper(buttons))
		End Sub
	End Class
End Namespace
